src/transport/SessionManager.h

```diff
@@ -249,9 +249,12 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate
         return mUnauthenticatedSessions.AllocInitiator(ephemeralInitiatorNodeID, peerAddress, config);
     }
 
-    // TODO: this is a temporary solution for legacy tests which use nodeId to send packets
-    // and tv-casting-app that uses the TV's node ID to find the associated secure session
-    SessionHandle FindSecureSessionForNode(NodeId peerNodeId);
+    //
+    // Find an existing secure session given a peer's scoped NodeId and a type of session to match against.
+    // If matching against all types of sessions is desired, kUnDefined should be passed into type.
+    //
+    void FindSecureSessionForNode(SessionHolder & sessionHolder, ScopedNodeId peerNodeId,
```

Comment:
Why is this taking a SessionHolder ref instead of returning a SessionHandle and lettting the caller do its own release/grab bits?

Reply:
We cannot return a `SessionHandle` since there's a possibility we may not find one.

So it would have to be `Optional<SessionHandle>`. `SessionHolder` permitted construction without a valid session, hence why I chose that, but I think that imparts the wrong semantic.

Will return `Optional<SessionHandle>` instead.